CMakeLists.txt

```diff
@@ -1,52 +1,62 @@
-cmake_minimum_required(VERSION 2.8)
+cmake_minimum_required( VERSION 3.2 )
+set( CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Supported configuration types" FORCE )
 
-project(stlab C CXX)
+project( stlab LANGUAGES C CXX )
 
-set(CMAKE_SKIP_RPATH OFF)
+option( stlab_playground "Configure and build the stlab playground" )
+option( coverage "Enable binary instrumentation to collect test coverage information in the DEBUG configuration" )
+option( stlab_testing "Compile the stlab tests and integrate with ctest" ${BUILD_TESTING} )
 
-if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")  
-  # The addition of "--coverage -fno-inline" is for lcov
-  #set(CMAKE_CXX_FLAGS "-g -Wall -O3 -std=c++14 -fsanitize=thread -fexceptions -D_GLIBCXX_USE_CXX11_ABI=0")
-  set(CMAKE_CXX_FLAGS "-O3 -g -Wall --coverage -fno-inline -std=c++14 -fexceptions -ftemplate-backtrace-limit=0 -D_GLIBCXX_USE_CXX11_ABI=0")
-  set(CMAKE_LINK_FLAGS "-pthread")
-  SET (CMAKE_SHARED_LINKER_FLAGS ${CMAKE_SHARED_LINKER_FLAGS_INIT} $ENV{LDFLAGS})
-endif()
-
-set (Boost_MULTITHREADED ON) 
-
-if(WIN32)
-  add_definitions(-D_WIN32_WINNT=0x0601)
-  set(Boost_USE_STATIC_LIBS ON)
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
-else()
-  set (Boost_USE_STATIC_LIBS ON)
-endif(WIN32)
+set( Boost_MULTITHREADED ON ) 
+set( Boost_USE_STATIC_LIBS ON )
 
-if(EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-    set(CONAN_SYSTEM_INCLUDES ON)
+if( EXISTS ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake )
+    include( ${CMAKE_BINARY_DIR}/conanbuildinfo.cmake )
+    set( CONAN_SYSTEM_INCLUDES ON )
     conan_basic_setup()
 else()
-    find_package(Boost 1.60.0 COMPONENTS unit_test_framework)
-    
-    include_directories(SYSTEM ${Boost_INCLUDE_DIR})
-    link_directories(${Boost_LIBRARY_DIR})
+    find_package( Boost 1.60.0 COMPONENTS unit_test_framework )
 endif()
 
-set (Boost_MULTITHREADED ON) 
-
-set(CMAKE_THREAD_PREFER_PTHREAD TRUE)
-
+set( CMAKE_THREAD_PREFER_PTHREAD TRUE )
 find_package( Threads )
 
-if(CMAKE_USE_PTHREADS_INIT)
-    set(CMAKE_C_FLAGS ${CMAKE_C_FLAGS} "-pthread")
+add_library( stlab INTERFACE )
+
+target_include_directories( stlab INTERFACE "${CMAKE_CURRENT_SOURCE_DIR}"
+                                  INTERFACE "${Boost_INCLUDE_DIRS}" )
+
+target_link_libraries( stlab INTERFACE ${CMAKE_THREAD_LIBS_INIT}
+                             INTERFACE ${CONAN_LIBS} )
+                           
+target_sources( stlab INTERFACE
```

Comment:
I'm a bit on the fence as well, to be honest. The documentation seems to frown on the practice

> We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate.

There is pretty good description of the drawbacks in Daniel Pfeifer's CppNow [presentation](https://youtu.be/bsXLMQ6WgIk?t=14m49s). In short, it's recommended for scripting mode rather than project mode.